src/openzeppelin/token/erc1155/IERC1155Receiver.cairo

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo v0.5.1 (token/erc1155/IERC1155Receiver.cairo)
+
+%lang starknet
+
+from starkware.cairo.common.uint256 import Uint256
+
+@contract_interface
+namespace IERC1155Receiver {
+    func onERC1155Received(
+        operator: felt, from_: felt, id: Uint256, amount: Uint256, data_len: felt, data: felt*
+    ) -> (selector: felt) {
+    }
+
+    func onERC1155BatchReceived(
+        operator: felt,
+        from_: felt,
+        ids_len: felt,
+        ids: Uint256*,
+        amounts_len: felt,
+        amounts: Uint256*,
+        data_len: felt,
+        data: felt*,
+    ) -> (selector: felt) {
+    }
+
+    func supportsInterface(interfaceId: felt) -> (success: felt) {
```

Comment:
i think the inconsistency reaches 721 too. they all should include the 165 functions.

Reply:
on a second thought, this is a bit more complicated. it would also imply to add all 721 methods to 721Metadata, heavily bloating the file. but on the other hand, not adding all of the methods would force users to import the "base interface" (be it 721, 1155, or 165) which would make a bad ux.

i think i lean towards bloating the interfaces.